shapeshift/web
# fix: defi earn grouping crypto amounts

## Description

Fixes the crypto amounts within the DeFi group earning dashboard, using the `getUnderlyingAssetIdsBalances` for underlying assets, and regular crypto amount divided by the exponent of the `AssetId` for top-level assets.

## Notice




## Pull Request Type


## Issue (if applicable)

N/A

## Risk



None as this isn't user-facing yet

## Testing



- Enable `DefiDashboard`
- Ensure crypto amounts are consistently shown and their decimal places are correct

### Engineering



_ ☝🏽 

### Operations



## Screenshots (if applicable)

#### Prod

<img width="1273" alt="image" src="https://user-images.githubusercontent.com/17035424/225887843-fd55bf79-ea1c-4f2a-bfc3-65a4c66d1c88.png">
<img width="1281" alt="image" src="https://user-images.githubusercontent.com/17035424/225888028-2b6bcb8a-f5e3-4d53-9730-972971c866e9.png">
<img width="1287" alt="image" src="https://user-images.githubusercontent.com/17035424/225888167-065ee3a8-ef6f-4bde-a253-e064addfde91.png">
<img width="1282" alt="image" src="https://user-images.githubusercontent.com/17035424/225888595-5d9f4d0b-6abc-4b49-aa76-7a04049b183a.png">

#### This diff

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/17035424/225887920-c1e80104-6dc4-4e6a-aa92-6f52a67dcd48.png">
<img width="1289" alt="image" src="https://user-images.githubusercontent.com/17035424/225888070-11cc7b4d-e34c-41af-a46d-73bd1791dc96.png">
<img width="1283" alt="image" src="https://user-images.githubusercontent.com/17035424/225888233-9002372d-1e53-414b-9d3a-2a5c53e0aa43.png">
<img width="1290" alt="image" src="https://user-images.githubusercontent.com/17035424/225888633-6c199837-4dfb-4563-9b9f-cfe96921c2c7.png">
